app/ui/legacy/src/main/java/com/fsck/k9/activity/AccountList.kt

```diff
@@ -0,0 +1,151 @@
+        }
+    }
+
+    /**
+     * Reload list of accounts when this activity is resumed.
+     */
+    public override fun onResume() {
+        super.onResume()
+        loadAccounts()
+    }
+
+    /**
+     * Cancel all running coroutines when this activity is paused.
+     */
+    public override fun onPause() {
+        super.onPause()
+        scope.cancel()
+    }
+
+    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+        val account = parent.getItemAtPosition(position) as BaseAccount
+        onAccountSelected(account)
+    }
+
+    /**
+     * Load accounts in a background thread and populate the list view with the results on main thread.
+     */
+    private fun loadAccounts() {
+        scope.launch {
+            val accounts = withContext(Dispatchers.IO) {
+                getPreferences().getAccounts()
+            }
+
+            populateListView(accounts)
+        }
+    }
+
+    /**
+     * Create a new [AccountsAdapter] instance and assign it to the [android.widget.ListView].
+     *
+     * @param realAccounts
+     * An array of accounts to display.
+     */
+    private fun populateListView(realAccounts: List<Account>) {
+        val accounts: MutableList<BaseAccount> = ArrayList()
+
+        if (isShowUnifiedInbox) {
+            val unifiedInboxAccount: BaseAccount = createUnifiedInboxAccount()
+            accounts.add(unifiedInboxAccount)
+        }
+
+        accounts.addAll(realAccounts)
+
+        listView.apply {
+            adapter = AccountsAdapter(accounts)
+            invalidate()
+        }
+    }
+
+    /**
+     * This method will be called when an account was selected.
+     *
+     * @param account
+     * The account the user selected.
+     */
+    protected abstract fun onAccountSelected(account: BaseAccount)
+
+    internal inner class AccountsAdapter(accounts: List<BaseAccount?>) : ArrayAdapter<BaseAccount?>(
+        this@AccountList,
+        0,
+        accounts,
+    ) {
+        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+            val account = getItem(position)
+            val view = convertView ?: layoutInflater.inflate(R.layout.accounts_item, parent, false)
+
+            val holder = (view.tag as? AccountViewHolder) ?: AccountViewHolder(view).apply {
+                view.tag = this
+            }
+
+            val accountName = account!!.name
+            if (accountName != null) {
+                holder.description.text = accountName
+                holder.email.text = account.email
+                holder.email.visibility = View.VISIBLE
+            } else {
+                holder.description.text = account.email
+                holder.email.visibility = View.GONE
+            }
+
+            if (account is Account) {
+                holder.chip.setBackgroundColor(account.chipColor)
+            } else {
+                holder.chip.setBackgroundColor(-0x666667)
```

Comment:
This could be

```kotlin
holder.chip.setBackgroundColor(0xff999999U.toInt())
```

or

```kotlin
holder.chip.setBackgroundColor(Color.argb(0xFF, 0x99, 0x99, 0x99))
```

Reply:
I didn't notice this strange conversion. I think this should be a color reference.